mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1150,17 +1172,11 @@ def _generate_model_predictions(self):
                 self.y_pred = np.concatenate(y_pred_list, axis=0)
             elif isinstance(sample_pred, list):
                 self.y_pred = sum(y_pred_list, [])
-            # handle if sample_pred is a pandas series
             elif isinstance(sample_pred, pd.Series):
                 self.y_pred = pd.concat(y_pred_list)
-            else:
```

Comment:
I'm curious why we want to remove raising this exception - in the else case is self.y_pred just not defined?